teammapper-backend/src/map/utils/tests/mapFactories.ts

```diff
@@ -20,6 +20,35 @@ export const createMmpMap = (overrides = {}): MmpMap => ({
     ...overrides
 })
 
+export const createMmpRootNode = (map: MmpMap, overrides = {}): MmpNode => ({
+    id: crypto.randomUUID(),
+    name: 'Root Node',
+    children: Array<MmpNode>(),
+    root: true,
+    coordinatesX: 1,
+    coordinatesY: 1,
+    ...overrides,
+    colorsName: "",
+    colorsBackground: "",
+    colorsBranch: "",
+    fontSize: 0,
+    fontStyle: "",
+    fontWeight: "",
+    imageSrc: "",
+    imageSize: 0,
+    linkHref: "",
+    locked: false,
+    detached: false,
+    k: 0,
+    nodeParentId: "",
+    orderNumber: 0,
+    lastModified: new Date(),
+    createdAt: new Date(),
+    nodeMap: map,
+    nodeMapId: map.id,
+    nodeParent: new MmpNode
```

Comment:
not sure I understand this? can you elaborate what his happening here?

Reply:
Regarding `nodeParent`? It's not nullable on either `Node` or `MmpNode` as far as TS is concerned, so there has to be _something_ there (in this case something very ugly :D)

Reply:
but its the root, how come its not nullable? We don't have parents for the root note itself?